### ISSUE TITLE
Demo very incomplete?

### ISSUE THREAD

**Issue by JacobStewart64** [BSVogler/WurfelEngineSDK#15]
Hi I built and ran the demo with much trouble with file paths and imports. Am I doing something wrong or is the demo very incomplete? No shooting, just a flat world. Weapons are selectable though. Monsters chase you but you can't kill them.

If that is all that is fine, I appreciate your work greatly. I hope to make a 2D mmo with this engine. If the demo should have more features, what am I doing wrong? I didn't use Maven, maybe that's it. I just cloned this repository.

Just a Java noob wondering what's going on lol

**Comment by JacobStewart64** [comment]
Oh yeah, the ground smashing thing, sometimes the ground gets smashed when I shoot, am I hitting invisible barrels? Am I doing something wrong or is the demo just not complete?

**Comment by BSVogler** [comment]
Hi, the demo is sadly in a very bad shape. There are some bugs like the ones you encountered. However, there shouldn't be so many problems with the imports as it uses maven. After an import in an IDE it should run from source code. Can you tell me more about the issues you had?

If you are a java noob then attempting to write an MMO is probably the wrong goal. Choose very small games and once you finished them attempt bigger ones.

**Comment by JacobStewart64** [comment]
I had auto-import on the fly turned off, I think it's turned off by default in IDEA, I spent quite a while manually doing imports before finding that option. Then it fixed them all for me. At the very top of the file, the package part, I had to add main.java. to each statement. Numerous imports, like class Enemy, were also missing from many of the files. Resource paths were all wrong, throwing errors when no resource could be found. I'm on Windows and I checked out your repo from SVN in IDEA. Also I had to manually set the main class. I am a Java noob, so it may just be me doing something dumb. I'm not a bad programmer though, I've implemented the groundwork of an MMO in C++ a few years ago. I figure Java will give me a leg up and I'll get a lot farther this time. Thank you for letting me know that it's not just me and the demo is not complete.

**Comment by JacobStewart64** [comment]
It turns out everything works nearly perfectly. I wasn't using Maven, I didn't know what it was lol. The only problem I'm having with running the demo while using maven is a simple fix really, but I kinda want to update the build so it truly works right out of the box. In weapon of choice pom the dependency wurfelengine-core-1.8.0.jar is wrong. I produced that jar by building the engine but it cannot be referenced by the demo pom like that. So I used a system dependency and got it working, only to find out that the jar doesn't include the gdx stuff. So your build system doesn't really work out of the box. Have to configure to use the wurfelengine-core-with-dependencies jar. I kinda want to try perfecting the build. I'll probably work on that. Also I want to dig into the pathfinding problem and see if I can solve that.
